helpers/user/current_user.go

```diff
@@ -0,0 +1,15 @@
+package helpers
+
+import (
+	"github.com/gin-gonic/gin"
+	"github.com/gutakk/go-google-scraper/models"
+)
+
+func GetCurrentUser(c *gin.Context) models.User {
+	currentUser := c.MustGet("currentUser")
+
+	if currentUser != nil {
+		return currentUser.(models.User)
```

Comment:
I am not sure why using [a type assertion](https://golang.org/ref/spec#Type_assertions) here as per the usage in the controller:

```go
currentUser := helpers.GetCurrentUser(c)
```

With the type assertion, shouldn't this be:

```go
currentUser, ok := helpers.GetCurrentUser(c)
```

Reply:
Because `MustGet("currentUser")` return interface type so I need to cast it to `models.User` type so I can use its attribute. As `currentUser` can be any type so I added the type check in 891f46f42d4ef5ab6165819c5744526a5bd28996

Reply:
Now with the type check, it does make more sense 😁